database.js

```diff
@@ -17,10 +18,22 @@ function init() {
 
 // Make sure that the database schema is up to date
 function ensureVersion() {
-    const query = "SHOW COLUMNS FROM mentors LIKE 'pmsg'";
-    connection.query(query, (err, rows) => {
+    // Add the current version to the "version" table if it is empty
+    const checkVer = "SHOW COLUMNS FROM version LIKE 'ver'";
+    connection.query(checkVer, (err, rows) => {
         if (err) throw err;
         if (rows.length === 0) {
+            const addVer = "INSERT INTO version VALUES (?)";
+            connection.query(addVer, [package.version], (err) => {
+                if (err) throw err;
+            });
+        }
+    });
+    // Detect older versions of the database
+    const query = "SELECT ver FROM version";
+    connection.query(query, (err, rows) => {
+        if (err) throw err;
+        if (rows[0] === "1.0.2") {
             // pmsg column not present
             console.log("Database version 1.0.2 detected. Migrating to version 1.0.3...");
```

Comment:
It is still necessary to check for the existence of the `pmsg` column. Course Connect versions 1.0.2 and 1.0.3 both do not have a `version` table, so databases from either of these versions may be flagged as version 1.0.2 (if my suggested change above is implemented). These changes should only be made if the database is truly 1.0.2; that is, the `pmsg` column is not present. Otherwise, the version number should just be incremented.